compiler/scenario-service/client/src/DA/Daml/LF/PrettyScenario.hs

```diff
@@ -492,6 +492,17 @@ prettyScenarioErrorError (Just err) =  do
         [ "Found duplicate contract keys in submitted disclosed contracts"
         , label_ "Template: " $ prettyMay "<missing template>" (prettyDefName world) templateId
         ]
+    ScenarioErrorErrorDisclosurePreprocessingNonExistentTemplate(ScenarioError_DisclosurePreprocessingNonExistentTemplate templateId) ->
```

Comment:
Why are you now including all the scenario stuff in this PR? I think it makes more sense to keep this focused on testing and address the rest separately. Otherwise, it becomes impossible to review it.

Reply:
As part of adding in a couple of ScenarioError classes for DisclosureProcessing (added to address the issue around template keys and disclosed contracts having no key hashes), I've had no option but to do this (the code would not build or ow compile).

Reply:
These errors should not be scenario errors. They are general engine errors.

Reply:
oh you mean exposing those entries in the scenario service?

Reply:
It appears that once you define new subclasses of com.daml.lf.interpretation.Error, you have effectively said that you want to expose those errors to the user (i.e. the scenario service in this case I guess?).

Reply:
makes sense, apologies, misread this change